src/map.rs

```diff
@@ -168,6 +168,19 @@ impl<'a, T> PutResult<'a, T> {
     }
 }
 
+/// The error type for the [`HashMap::try_insert`] method.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct TryInsertError<'a, V> {
+    /// A reference to the old value mapped to the key.
+    pub old: &'a V,
+    /// A reference to the new value, the one passed as an argument to
+    /// [`HashMap::try_insert`].
+    pub not_inserted: &'a V,
```

Comment:
This should be a `V`, not a `&'a V`.

Reply:
(Just dereference the `Box<V>` to move the `V` out of there).

Reply:
Alright.

Reply:
I can't remember how to unwrap a `Box` hehe. Couldn't find it in the docs either, would you mind helping me out?

Also, has the `box` keyword being stabilized yet? If it is, I think I can just patter match on `box contents_of_the_box`, can't I?

Reply:
> (Just dereference the `Box<V>` to move the `V` out of there).

Has the `box` keyword been stabilized already? I think I could just pattern match on `box inner_contents_of_the_box` if that's the case, couldn't I?

Reply:
No, I don't think so. Last I heard it also wasn't going to stabilize any time soon. Does `*not_inserted` not work?

Reply:
It works. That's the solution I went for. I was just asking because the `box` thing seemed more ergonomic and, as listener of _Rustacian Station_, I knew you'd probably know the answer to my question ❤️